src/homework01/Tasks.kt

```diff
@@ -1,13 +1,228 @@
 package homework01
 
-fun binarySearch(array: List<Int>, f: Int): Int = 0
+/**
+ * -1 is added when value not found
+ * to avoid ambiguous results when value
+ * could've been inserted to 0th index
+ *
+ * @see binarySearchRecursion
+ */
+fun linearSearch(array: List<Int>, element: Int, start: Int, end: Int, compare: (Int, Int) -> Int): Int {
+    for (i in start..end) {
+        if (compare(array[i], element) == 0) {
+            return i
+        }
+        if (compare(array[i], element) > 0) {
+            return -i - 1
+        }
+    }
+    return -end - 1
+}
 
-fun howManyNumbers(array: List<Int>, l: Int, r: Int): Int = 0
+/**
+ * start and end are assumed to be in [0..array.size]
+ * also assumed that start <= end
+ * array is assumed to be sorted in ascending order according to comparator
+ *
+ * @param compare thr comparator to be used.
+ *        Should return negative value if first argument is smaller,
+ *        positive value if first argument is greater
+ *        and 0 if they are equal
+ *
+ * @returns index of first occurrence of element
+ */
+fun binarySearchRecursion(array: List<Int>, element: Int, start: Int, end: Int, compare: (Int, Int) -> Int): Int {
 
-fun getSumOfPrime(k: Int): Long = 0
+    // At some point recursive splitting is no longer that profitable
+    if (end - start < 5) {
```

Comment:
It's better to make these _magical literals_ -> _magical constants_ and move them _Somewhere over the rainbow (way up high)_ :star2: